src/webgpu/shader/execution/expression/binary/bool_logical.spec.ts

```diff
@@ -0,0 +1,98 @@
+export const description = `
+Execution Tests for the boolean binary logical expression operations
+`;
+
+import { makeTestGroup } from '../../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../../gpu_test.js';
+import { bool, TypeBool } from '../../../../util/conversion.js';
+import { allInputSources, run } from '../expression.js';
+
+import { binary } from './binary.js';
+
+export const g = makeTestGroup(GPUTest);
+
+g.test('and')
+  .specURL('https://www.w3.org/TR/WGSL/#logical-expr')
+  .desc(
+    `
+Expression: e1 & e2
+Logical "and". Component-wise when T is a vector. Evaluates both e1 and e2.
```

Comment:
Can we add a test that this doesn't short circuit? Like call a function with side effect in both sides and show both side effects happened?

Reply:
Done, in src/webgpu/shader/execution/evaluation_order.spec.ts